scala-package/core/src/main/scala/org/apache/mxnet/IO.scala

```diff
@@ -136,11 +142,29 @@ class DataBatch(val data: IndexedSeq[NDArray],
                 val pad: Int,
                 // the key for the bucket that should be used for this batch,
                 // for bucketing io only
-                val bucketKey: AnyRef = null,
+                val bucketKey: AnyRef,
                 // use ListMap to indicate the order of data/label loading
                 // (must match the order of input data/label)
-                private val providedData: ListMap[String, Shape] = null,
-                private val providedLabel: ListMap[String, Shape] = null) {
+                private val providedData: ListMap[String, Shape],
+                private val providedLabel: ListMap[String, Shape],
+                val dataDType: DType,
+                val labelDType: DType,
+                val dataLayout: String,
+                val labelLayout: String) {
```

Comment:
I'm wondering whether we can pack them all into dataDesc and labelDesc?

Reply:
how about replace `providedData` by `ListMap[String, DataDesc]`, and providedLabel respectively. (can create a new constructor to keep backward-compatibility.
Actually there's no guarantee that all inputs share the same layout and dtype.

Reply:
@nswamy maybe need to keep the `Map`, I think some usage queries by name.

Reply:
can you elaborate a bit? DataDesc already has `name`